src/handlers/events/delivery/scan/review/OnPendingAutofixStatus.ts

```diff
@@ -0,0 +1,115 @@
+import { chainEditors } from "@atomist/automation-client/operations/edit/projectEditorOps";
+import { GitCommandGitProject } from "@atomist/automation-client/project/git/GitCommandGitProject";
+import { Goal } from "../../../../../common/goals/Goal";
+import { OnAnyPendingStatus, StatusState } from "../../../../../typings/types";
+import { createStatus } from "../../../../../util/github/ghub";
+import { ContextToPlannedPhase } from "../../goals/httpServiceGoals";
+import { forApproval } from "../../verify/approvalGate";
+
+/**
+ * Run any autofix editors on a push.
+ * Set GitHub success status
+ */
+@EventHandler("Make autofixes",
+    GraphQL.subscriptionFromFile("graphql/subscription/OnAnyPendingStatus.graphql"))
+export class OnPendingAutofixStatus implements HandleEvent<OnAnyPendingStatus.Subscription> {
+
+    @Secret(Secrets.OrgToken)
+    private githubToken: string;
+
+    private editorChain: ProjectEditor;
+
+    constructor(public goal: Goal,
+                private editors: AnyProjectEditor[] = []) {
+        this.editorChain = editors.length > 0 ? chainEditors(...editors) : undefined;
+    }
+
+    public async handle(event: EventFired<OnAnyPendingStatus.Subscription>,
+                        context: HandlerContext,
+                        params: this): Promise<HandlerResult> {
+        const status = event.data.Status[0];
+        const commit = status.commit;
+
+        const id = new GitHubRepoRef(commit.repo.owner, commit.repo.name, commit.sha);
+        const credentials = {token: params.githubToken};
+
+        if (status.context !== params.goal.context || status.state !== "pending") {
+            logger.warn(`I was looking for ${params.goal.context} being pending, but I heard about ${status.context} being ${status.state}`);
+            return Success;
+        }
+
+        logger.info("Will apply %d autofixes to %j", params.editors.length, id);
+        try {
+            const project = await GitCommandGitProject.cloned(credentials, id);
+
+            if (params.editorChain) {
+                // TODO parameterize this
+                const editMode: BranchCommit = {
+                    branch: commit.pushes[0].branch,
+                    message: "Autofixes",
+                };
+                logger.info("Editing %j with mode=%j", id, editMode);
+                await editOne(context, credentials, params.editorChain, editMode,
+                    new SimpleRepoId(id.owner, id.repo));
+            }
+
+            await markScanned(project.id as GitHubRepoRef,
+                params.goal.context, "success", credentials, false);
+
+            return Success;
+        } catch (err) {
+            await markScanned(id,
+                params.goal.context, "error", credentials, false);
+            return failure(err);
+        }
+    }
+}
+
+// TODO fix this
+export const ScanBase = "https://scan.atomist.com";
```

Comment:
I think we should not link to anything. These don't require approval, it doesn't make sense to require approval for an autofix step that either does nothing or supersedes the commit.